testcode/lib/em-test.c

```diff
@@ -85,9 +117,27 @@ int main (void)
     unsigned PageCount;
     unsigned char X, Y;
     struct em_copy c;
+    unsigned index;
+    signed char valid_key = -1;
+    char key;
+
+    clrscr ();
+    cputs ("Which RAM exp to test?\r\n\r\n");
+    for (index = 0; drivers[index].displayname; ++index) {
+        cprintf("%c: %s\r\n", drivers[index].key, drivers[index].displayname);
+    }
+
+    while (valid_key == -1) {
```

Comment:
thank you for all your comments, grek.

I'm not used to compiling for a small memory footprint, I can see how using the key member (byte) would save (code) space and (execution) time, and I can see how < 0 might translate to something like a 'tst' opcode instead of a 'cmp' opcode, which would again save.